src/library_proxyfs.js

```diff
@@ -192,6 +192,35 @@ mergeInto(LibraryManager.library, {
           throw new FS.ErrnoError(ERRNO_CODES[e.code]);
         }
       },
+      // NB: can't delegate mmap call from PROXYFS to underlying FS,
+      // because it may originate from another module,
+      // and then it'll use mmapAlloc of that module
+      // so the memory will be filled with file in that module, not in our module
+      // Instead, we should allocate module here on our own and fill it with the content
+      mmap: function(stream, length, position, prot, flags) {
+        var path = PROXYFS.realPath(stream.node);
+        var contents = stream.node.mount.opts.fs.readFile(path);
+
+        if (!(flags & 2)) {
+          throw new Error("Not implemented MAP_SHARED (yet)");
```

Comment:
I think the flags should already have been checked by `mmap` in `library_fs.js`, so we shouldn't need to check them again here.

Reply:
Thank you, I copied this from the existing mmap.

Reply:
I don't think we should merge code if we don't know how it works or what it's for. If you remove the flag checking here and  try to pass MAP_SHARED, is it caught by previous flag checking?